src/pages/settings/Subscription/PaymentCard/ChangeBillingCurrency/index.tsx

```diff
@@ -19,11 +20,23 @@ function ChangeBillingCurrency() {
     const [fundList] = useOnyx(ONYXKEYS.FUND_LIST);
     const defaultCard = useMemo(() => Object.values(fundList ?? {}).find((card) => card.accountData?.additionalData?.isBillingCard), [fundList]);
 
-    const changeBillingCurrency = useCallback((currency?: ValueOf<typeof CONST.PAYMENT_CARD_CURRENCY>, values?: FormOnyxValues<typeof ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM>) => {
-        if (values?.securityCode) {
-            PaymentMethods.updateBillingCurrency(currency ?? CONST.PAYMENT_CARD_CURRENCY.USD, values.securityCode);
+    const [formData] = useOnyx(ONYXKEYS.FORMS.CHANGE_BILLING_CURRENCY_FORM);
+    const formDataComplete = formData?.isLoading === false && !formData.errors;
+    const prevFormDataComplete = usePrevious(formDataComplete);
+
+    useEffect(() => {
```

Comment:
Bug: saving twice closes the modal.

1. change currency
2. enter wrong cvv > save
3. save again

https://github.com/user-attachments/assets/7f6f4a92-bda7-430a-af48-3b4cca2f28b6

Reply:
@rushatgabhane
As far as I know, 123 is a valid CVV, and this bug might occur due to incorrect setting steps.
I have asked about these steps and will check again when I have the answer.

Reply:
Depends on what card @rushatgabhane was using. @rushatgabhane were you using a [test card from stripe](https://docs.stripe.com/testing#cards) or were you using a real credit card that has a real cvv code? Also would be curious to see the network tab to see if the error gets any more specific than `"Auth SetFundCurrency" returned an error"`

Reply:
I was using a card set using `useEffect`

`Onyx.merge(cardList, 1:{...})`.


But that shouldn't matter. The frontend should call the API again and on the second save and wait for the API response.

Reply:
Sounds good, I will DM you @rushatgabhane with credentials to my ngrok so that you can also test this using stripe test cards if you'd like.

Reply:
@rushatgabhane Here is the result when I use [test card from stripe](https://docs.stripe.com/testing#cards) and env from @blimpich 

https://github.com/user-attachments/assets/48cd9e07-68e3-473f-95c4-90f57e75e198

Reply:
@nkdengineer those steps are not correct because you're not testing when the API response has an error 


1. Make the API return non status 200
2. Enter a 3 digit cvv
3. Tap submit, observe error message above button 
4. Tap submit again

Reply:
Can we keep this moving, please?

Reply:
I'll update today

Reply:
> Make the API return non status 200

I tried many ways including turning off validate but BackEnd always returns 200 when I click save with env from blimpich

> I was using a card set using useEffect
Onyx.merge(cardList, 1:{...})

@rushatgabhane please add more detail here so I can reproduce this bug

Reply:
@nkdengineer you can look at the typescript file for cardList and fill in card number etc

Reply:
@rushatgabhane i fixed this bug, please check again

https://github.com/user-attachments/assets/346d57e9-40fc-4160-a52d-b15d7280c807

Reply:
@rushatgabhane friendly bump